### PR TITLE
wan i2v and t2v

### PR DESCRIPTION
**Pull request by alexarmbr**

### CODE REVIEW

File: src/para_attn/context_parallel/diffusers_adapters/wan_video.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+            padding_length = padded_seq_length - seq_length
+
+            b, c, s, d = rotary_emb.shape
+            rotary_emb = torch.cat(
+                [rotary_emb, torch.zeros(b, c, padding_length, d, device=rotary_emb.device, dtype=rotary_emb.dtype)],
+                dim=2,
+            )
+
+            b, s, d = hidden_states.shape
+            hidden_states = torch.cat(
```
Comment (chengzeyi) [inline src/para_attn/context_parallel/diffusers_adapters/wan_video.py:83]: I am afraid that padding directly is not really suitable because these parts will still contribute to the attention score. So we might need to find better ways to deal with that.

**Review by chengzeyi — CHANGES_REQUESTED** [review]

File: src/para_attn/context_parallel/diffusers_adapters/wan_video.py (hunk 1 above)
Comment (alexarmbr) [reply src/para_attn/context_parallel/diffusers_adapters/wan_video.py:83]: what if we pad with `-inf` so those values will go to 0 in the softmax, like is done with causal masking?

Comment (chengzeyi) [reply src/para_attn/context_parallel/diffusers_adapters/wan_video.py:83]: @alexarmbr I think it is going to be more complicated than that. SparseKVAttnMode is one solution for that.

Comment (chengzeyi) [reply src/para_attn/context_parallel/diffusers_adapters/wan_video.py:83]: If I2V cannot work with all shapes, we can resize the image outside.

File: parallel_examples/run_wan.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+pipe.scheduler = UniPCMultistepScheduler.from_config(pipe.scheduler.config, flow_shift=3.0)
+pipe.to("cuda")
+
+from para_attn.context_parallel import init_context_parallel_mesh
+from para_attn.context_parallel.diffusers_adapters import parallelize_pipe
+
+mesh = init_context_parallel_mesh(pipe.device.type)
+parallelize_pipe(pipe, mesh=mesh)
+
+torch._inductor.config.reorder_for_compute_comm_overlap = True
```
[the remark this answers is not in the document]
Comment (alexarmbr) [reply parallel_examples/run_wan.py:26]: this is slightly faster, I will add it

File: parallel_examples/run_wan.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+torch.set_float32_matmul_precision("high")
```
[the remark this answers is not in the document]
Comment (alexarmbr) [reply parallel_examples/run_wan.py:27]: not just for float32 vae, there are other parts of the model that run in float32 like [here](https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/transformers/transformer_wan.py#L113)

File: parallel_examples/run_wan.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+from diffusers.schedulers.scheduling_unipc_multistep import UniPCMultistepScheduler
+
+from diffusers.utils import export_to_video
+
+dist.init_process_group()
+
+torch.cuda.set_device(dist.get_rank())
+
+model_id = "Wan-AI/Wan2.1-T2V-14B-Diffusers"
+vae = AutoencoderKLWan.from_pretrained(model_id, subfolder="vae", torch_dtype=torch.float32)
```
[the remark this answers is not in the document]
Comment (chengzeyi) [reply parallel_examples/run_wan.py]: @chengzeyi I see but I will prefer bfloat16 for memory savings.

File: parallel_examples/run_wan.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (chengzeyi) [reply parallel_examples/run_wan.py]: @alexarmbr Got it
